pkg/resource/logging-credentials/logging_operator_secrets.go

```diff
@@ -0,0 +1,75 @@
+package loggingcredentials
+
+import (
+	"math/rand"
+	"strings"
+	"time"
+
+	loggedcluster "github.com/giantswarm/logging-operator/pkg/logged-cluster"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// LoggingCredentialsSecretMeta returns metadata for the logging-operator credentials secret.
+func LoggingCredentialsSecretMeta(lc loggedcluster.Interface) metav1.ObjectMeta {
+	return metav1.ObjectMeta{
+		Name:      lc.AppConfigName("logging-credentials"),
+		Namespace: "monitoring",
+	}
+}
+
+// Generate a random 20-characters password
+func genPassword() string {
+	const length = 20
+
+	rand.Seed(time.Now().UnixNano())
+	chars := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
+		"abcdefghijklmnopqrstuvwxyz" +
+		"0123456789" +
+		"~=+%^*/()[]{}/!@#$?|")
+
+	var b strings.Builder
+	for i := 0; i < length; i++ {
+		b.WriteRune(chars[rand.Intn(len(chars))])
+
+	}
+	str := b.String() // E.g. "ExcbsVQs"
+	return str
+}
+
+// GenerateObservabilityBundleConfigMap returns a configmap for
+// the observabilitybundle application to enable promtail.
+func GenerateLoggingCredentialsBasicSecret(lc loggedcluster.Interface) (v1.Secret, error) {
+
+	secret := v1.Secret{
+		ObjectMeta: LoggingCredentialsSecretMeta(lc),
+		Data:       map[string][]byte{},
+	}
+
+	return secret, nil
+}
+
+// Update a LoggingCredentials secret if needed
+func UpdateLoggingCredentials(loggingCredentials v1.Secret) bool {
```

Comment:
Have you tested this ? I think you need to pass loggingCredentials as a pointer here if you want to update it in place.

Reply:
Short version: Yes, tested and it updates in place.

Longer version: I thought like you and did something like `newversion := UpdateLoggingCredentials(oldversion)` and took way longer than I should debugging why it didn't see differences between `newversion` and `oldversion` :rofl:

Reply:
Updated it to use a pointer.